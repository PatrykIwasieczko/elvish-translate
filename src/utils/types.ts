export type Translation = {
  id: string;
  english: string;
  elvish: string;
};
