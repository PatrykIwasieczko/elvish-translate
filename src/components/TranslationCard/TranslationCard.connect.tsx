import { useState } from "react";
import { TranslationCardConnectProps } from "./TranslationCard.interface";
import { useAppDispatch } from "../../store/store";
import { TranslationCard } from "./TranslationCard";
import { deleteFavourite, editFavourite } from "../../store/actions";

export const TranslationCardConnect: React.FC<TranslationCardConnectProps> = ({
  translation,
}) => {
  const [isEditing, setIsEditing] = useState(false);
  const [editedTranslation, setEditedTranslation] = useState(translation);
  const dispatch = useAppDispatch();

  const handleEdit = () => {
    dispatch(editFavourite(editedTranslation));
    setIsEditing(false);
  };

  const handleDelete = () =>
    isEditing ? setIsEditing(false) : dispatch(deleteFavourite(translation.id));

  return (
    <TranslationCard
      translation={translation}
      isEditing={isEditing}
      setIsEditing={setIsEditing}
      editedTranslation={editedTranslation}
      setEditedTranslation={setEditedTranslation}
      onEdit={handleEdit}
      onDelete={handleDelete}
    />
  );
};
