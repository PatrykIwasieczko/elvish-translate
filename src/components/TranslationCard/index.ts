export { TranslationCard } from "./TranslationCard";
export { TranslationCardConnect } from "./TranslationCard.connect";
