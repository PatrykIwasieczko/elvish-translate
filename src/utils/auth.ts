import Cookies from "js-cookie";

export const login = (token: string) => {
  Cookies.set("user", token, { expires: 1 });
};

export const logout = () => {
  Cookies.remove("user");
};

export const isAuthenticated = () => {
  return !!Cookies.get("user");
};
