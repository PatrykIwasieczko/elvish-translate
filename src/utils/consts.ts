export const TRANSLATION = "translation";
export const FAVOURITE_TRANSLATIONS = "favouriteTranslations";

export const FETCH_TRANSLATION = "fetchTranslation";
export const ADD_FAVOURITE = "addFavourite";
export const LOAD_FAVOURITES = "loadFavourites";
export const DELETE_FAVOURITE = "deleteFavourites";
export const EDIT_FAVOURITE = "EDIT_FAVOURITE";
