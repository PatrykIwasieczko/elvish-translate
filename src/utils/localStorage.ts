export const saveLocalStorageItems = (
  id: string,
  items: unknown[] | unknown
) => {
  try {
    const serializedState = JSON.stringify(items);
    localStorage.setItem(id, serializedState);
  } catch (error) {
    console.error("Could not save state to localStorage", error);
  }
};

export const getLocalStorageItems = (id: string) => {
  try {
    const serializedState = localStorage.getItem(id);
    if (serializedState === null) {
      return null;
    }
    return JSON.parse(serializedState);
  } catch (error) {
    console.error("Could not load state from localStorage", error);
    return null;
  }
};
