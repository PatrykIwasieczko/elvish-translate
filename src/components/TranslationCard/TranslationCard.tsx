import { TranslationCardProps } from "./TranslationCard.interface";

export const TranslationCard: React.FC<TranslationCardProps> = ({
  isEditing,
  setIsEditing,
  editedTranslation,
  setEditedTranslation,
  translation,
  onEdit,
  onDelete,
}) => {
  return (
    <div className="max-w-md mx-auto bg-white shadow-lg rounded-lg my-4 min-h-[200px] w-full flex flex-col justify-between">
      <div className="px-6 py-4">
        {isEditing ? (
          <div className="text-gray-800 text-lg mb-4">
            <p>
              <strong>English:</strong>{" "}
              <input
                value={editedTranslation.english}
                onChange={(e) =>
                  setEditedTranslation({
                    ...editedTranslation,
                    english: e.target.value,
                  })
                }
                className="w-full px-3 py-2 border border-gray-300 rounded-md mb-4"
              />
            </p>
            <p>
              <strong>Elvish:</strong>{" "}
              <input
                value={editedTranslation.elvish}
                onChange={(e) =>
                  setEditedTranslation({
                    ...editedTranslation,
                    elvish: e.target.value,
                  })
                }
                className="w-full px-3 py-2 border border-gray-300 rounded-md mb-4"
              />
            </p>
          </div>
        ) : (
          <div className="text-gray-800 text-lg mb-4">
            <p className="break-words">
              <strong>English:</strong> {translation.english}
            </p>
            <p className="break-words">
              <strong>Elvish:</strong> {translation.elvish}
            </p>
          </div>
        )}
      </div>

      <div className="px-6 py-4 border-t border-gray-200 flex justify-between">
        {isEditing ? (
          <button
            onClick={onEdit}
            className="bg-blue-500 text-white px-4 py-2 rounded-md hover:bg-blue-600"
          >
            Save
          </button>
        ) : (
          <button
            onClick={() => setIsEditing(true)}
            className="bg-gray-500 text-white px-4 py-2 rounded-md hover:bg-gray-600"
          >
            Edit
          </button>
        )}

        <button
          onClick={onDelete}
          className="bg-red-500 text-white px-4 py-2 rounded-md hover:bg-red-600"
        >
          {isEditing ? "Cancel" : "Delete"}
        </button>
      </div>
    </div>
  );
};
