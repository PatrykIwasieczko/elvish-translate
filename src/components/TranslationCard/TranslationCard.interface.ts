import { Translation } from "../../utils/types";

export interface TranslationCardConnectProps {
  translation: Translation;
}

export interface TranslationCardProps {
  translation: Translation;
  isEditing: boolean;
  setIsEditing: (isEditing: boolean) => void;
  editedTranslation: Translation;
  setEditedTranslation: (editedTranslation: Translation) => void;
  onEdit: () => void;
  onDelete: () => void;
}
