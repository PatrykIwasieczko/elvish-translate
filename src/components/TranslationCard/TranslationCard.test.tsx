import { render } from "@testing-library/react";
import { TranslationCard } from "./TranslationCard";
import { Translation } from "../../utils/types";

describe("components/TranslationCard", () => {
  const mockedTranslation: Translation = {
    id: "1",
    english: "English text",
    elvish: "Elvish text",
  };

  it("should render TranslationCard component", () => {
    const { container } = render(
      <TranslationCard
        translation={mockedTranslation}
        isEditing={false}
        setIsEditing={jest.fn()}
        editedTranslation={mockedTranslation}
        setEditedTranslation={jest.fn()}
        onEdit={jest.fn()}
        onDelete={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should render TranslationCard component in edit mode", () => {
    const { container } = render(
      <TranslationCard
        translation={mockedTranslation}
        isEditing={true}
        setIsEditing={jest.fn()}
        editedTranslation={mockedTranslation}
        setEditedTranslation={jest.fn()}
        onEdit={jest.fn()}
        onDelete={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
