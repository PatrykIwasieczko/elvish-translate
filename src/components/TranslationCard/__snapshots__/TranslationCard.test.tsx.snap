// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TranslationCard should render TranslationCard component 1`] = `
<div>
  <div
    class="max-w-md mx-auto bg-white shadow-lg rounded-lg my-4 min-h-[200px] w-full flex flex-col justify-between"
  >
    <div
      class="px-6 py-4"
    >
      <div
        class="text-gray-800 text-lg mb-4"
      >
        <p
          class="break-words"
        >
          <strong>
            English:
          </strong>
           
          English text
        </p>
        <p
          class="break-words"
        >
          <strong>
            Elvish:
          </strong>
           
          Elvish text
        </p>
      </div>
    </div>
    <div
      class="px-6 py-4 border-t border-gray-200 flex justify-between"
    >
      <button
        class="bg-gray-500 text-white px-4 py-2 rounded-md hover:bg-gray-600"
      >
        Edit
      </button>
      <button
        class="bg-red-500 text-white px-4 py-2 rounded-md hover:bg-red-600"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;

exports[`components/TranslationCard should render TranslationCard component in edit mode 1`] = `
<div>
  <div
    class="max-w-md mx-auto bg-white shadow-lg rounded-lg my-4 min-h-[200px] w-full flex flex-col justify-between"
  >
    <div
      class="px-6 py-4"
    >
      <div
        class="text-gray-800 text-lg mb-4"
      >
        <p>
          <strong>
            English:
          </strong>
           
          <input
            class="w-full px-3 py-2 border border-gray-300 rounded-md mb-4"
            value="English text"
          />
        </p>
        <p>
          <strong>
            Elvish:
          </strong>
           
          <input
            class="w-full px-3 py-2 border border-gray-300 rounded-md mb-4"
            value="Elvish text"
          />
        </p>
      </div>
    </div>
    <div
      class="px-6 py-4 border-t border-gray-200 flex justify-between"
    >
      <button
        class="bg-blue-500 text-white px-4 py-2 rounded-md hover:bg-blue-600"
      >
        Save
      </button>
      <button
        class="bg-red-500 text-white px-4 py-2 rounded-md hover:bg-red-600"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
